tests/test_server/mod.rs

```diff
@@ -0,0 +1,29 @@
+
+use std::process::Child;
+
+pub struct ExampleServer(Child);
+
+impl std::ops::Drop for ExampleServer {
+    fn drop(&mut self) {
+        let Ok(_) = self.0.kill()else{
+            println!("faild kill a process. ");
+            return;
+        };
+    }
+}
+
+pub fn run_example_server(example_name: &str, port_number: usize) -> ExampleServer {
+    let temp = assert_fs::TempDir::new().unwrap();
+    ExampleServer(
+        escargot::CargoBuild::new()
+            .example(example_name)
+            .manifest_path("Cargo.toml")
+            .target_dir(temp.path())
+            .run()
+            .unwrap()
+            .command()
+            .arg(port_number.to_string())
```

Comment:
As I think it's easier and just as fine to just hardcode the port numbers as unique ones, starting with the first example with port `40_000`, there will be no need to pass in a port number.